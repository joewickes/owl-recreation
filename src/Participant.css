* {
  margin: 0px;
  padding: 0px;
}

.Participant {
  display: flex;
  align-items: center;
}

.participant-left {
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid black;
  border-radius: 50%;
  width: 40px;
  margin: 10px;
}

.participant-left img {
  width: 100%;
  border-radius: 50%;
}

.participant-right {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  height: 80%;
  justify-content: flex-end;
}

.participant-name {
  font-weight: bold;
  margin-bottom:  5px;
}

.session-status {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  font-weight: lighter;
}

.session-green, .session-red {
  width: 9px;
  height: 9px;
  border-radius: 50%;
  margin-right: 10px;
}

.session-green {
  background-color: lightgreen;
}

.session-red {
  background-color: red;
}