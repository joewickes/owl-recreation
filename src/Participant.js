import React from 'react';
import './Participant.css';

class Participant extends React.Component {
  static defaultProps = {
    name: '',
    avatar: '',
    inSession: false,
    onStage: false
  }

  render() {
    const {name, avatar, inSession, onStage} = this.props;

    let sessionStatus = null;
    
    if (onStage) {
      // Show green light and 'on stage'
      sessionStatus = (
        <div className="session-status">
          <div className="session-green">
            <p></p>
          </div>
          <p>on stage</p>
        </div>
      );
    } else if (inSession) {
      // Show green light and 'in session'
      sessionStatus = (
        <div className="session-status">
          <div className="session-green">
            <p></p>
          </div>
          <p>in session</p>
        </div>
      );
    } else if (!inSession) {
      sessionStatus = (
        <div className="session-status">
          <div className="session-red">
            <p></p>
          </div>
          <p>left</p>
        </div>
      );
    }

    return (
      <div className="Participant">
        <div className="participant-left">
          <img src={avatar} alt={`${name}'s Avatar`} />
        </div>
        <div className="participant-right">
          <p className="participant-name">{name}</p>
          {sessionStatus}
        </div>
      </div>
    );
  }
  
}

export default Participant;