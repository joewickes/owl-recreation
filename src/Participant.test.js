import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';

describe('Participant Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the Participant on stage', () => {
    const tree = renderer
      .create(<Participant name="On Stage" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={true} onStage={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Participant in session', () => {
    const tree = renderer
      .create(<Participant name="In Session" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={true} onStage={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the Participant having left', () => {
    const tree = renderer
      .create(<Participant name="They Left" avatar={'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1'} inSession={false} onStage={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});