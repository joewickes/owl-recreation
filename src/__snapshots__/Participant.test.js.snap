// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Component renders the Participant having left 1`] = `
<div
  className="Participant"
>
  <div
    className="participant-left"
  >
    <img
      alt="They Left's Avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-right"
  >
    <p
      className="participant-name"
    >
      They Left
    </p>
    <div
      className="session-status"
    >
      <div
        className="session-red"
      >
        <p />
      </div>
      <p>
        left
      </p>
    </div>
  </div>
</div>
`;

exports[`Participant Component renders the Participant in session 1`] = `
<div
  className="Participant"
>
  <div
    className="participant-left"
  >
    <img
      alt="In Session's Avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-right"
  >
    <p
      className="participant-name"
    >
      In Session
    </p>
    <div
      className="session-status"
    >
      <div
        className="session-green"
      >
        <p />
      </div>
      <p>
        in session
      </p>
    </div>
  </div>
</div>
`;

exports[`Participant Component renders the Participant on stage 1`] = `
<div
  className="Participant"
>
  <div
    className="participant-left"
  >
    <img
      alt="On Stage's Avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
  <div
    className="participant-right"
  >
    <p
      className="participant-name"
    >
      On Stage
    </p>
    <div
      className="session-status"
    >
      <div
        className="session-green"
      >
        <p />
      </div>
      <p>
        on stage
      </p>
    </div>
  </div>
</div>
`;
